### PR TITLE
Fixed the issue with SCHEMA object being imported in case of --post-import-data

### PR DESCRIPTION
**Pull request by ShivanshGahlot**
as well and indexes from multiple schemas failing to import.

- Changed a condition in `applySchemaObjectFilterFlags` function in `importSchemaYugabyteDB.go` to fix the issue with SCHEMA object being imported in case of --post-import-data as well.
- Made multiple changes in` importData.go` to extract the schema related to the index name and add it to the index name before trying dropping the index with the mentioned index name.

### CODE REVIEW

File: yb-voyager/cmd/importData.go [hunk 1]
```diff
@@ -1035,8 +1062,15 @@ func executeSqlStmtWithRetries(conn **pgx.Conn, sqlInfo sqlInfo, objType string)
 			// creating fresh connection
 			(*conn).Close(context.Background())
 			*conn = newTargetConn()
+
+			// Extract the schema name and add to the index name
+			fullyQualifiedObjName, err := getIndexName(sqlInfo.stmt, sqlInfo.objName)
+			if err != nil {
+				utils.ErrExit("unable to get index name qualified with schema name for DDL %v: %w", sqlInfo.stmt, err)
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:1069]: Be precise and concise about the messages:

    "extract qualified index name from DDL [%v]: %w", sqlInfo.stmt, err

File: yb-voyager/cmd/importData.go [hunk 2]
```diff
@@ -1010,6 +1010,33 @@ func executeSqlFile(file string, objType string, skipFn func(string, string) boo
...
+		err := fmt.Errorf("not a CREATE INDEX statement")
+		return "", err
+	}
+
+	// Return the index name itself if it is aleady qualified with schema name
+	if len(strings.Split(indexName, ".")) == 2 {
+		return indexName, nil
+	}
+
+	fullyQualifiedIndexName := indexName
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:1025]: Not required.

File: yb-voyager/cmd/importData.go [hunk 3]
```diff
@@ -1010,6 +1010,33 @@ func executeSqlFile(file string, objType string, skipFn func(string, string) boo
...
+
+	splits := strings.Fields(sqlQuery)
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:1027]: Use `strings.FieldsFunc()`.
You can rename `splits` as `parts` or `words`.

File: yb-voyager/cmd/importData.go [hunk 4]
```diff
@@ -1010,6 +1010,33 @@ func executeSqlFile(file string, objType string, skipFn func(string, string) boo
...
+
+	for index, split := range splits {
+		if strings.EqualFold(split, "ON") {
+			tableName := splits[index+1]
+			schemaName := getTargetSchemaName(tableName)
+			fullyQualifiedIndexName = schemaName + "." + indexName
+			break
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:1034]: return fmt.Sprintf("%s.%s", schemaName, indexName), nil

File: yb-voyager/cmd/importData.go [hunk 5]
```diff
@@ -1010,6 +1010,33 @@ func executeSqlFile(file string, objType string, skipFn func(string, string) boo
...
+		}
+	}
+	return fullyQualifiedIndexName, nil
```
Comment (amit-yb) [inline yb-voyager/cmd/importData.go:1037]: return "", fmt.Errorf("could not find `ON` keyword in the CREATE INDEX statement")

**Review by amit-yb — COMMENTED** [review]
Overall looks good. Just a few minor comments.

File: yb-voyager/cmd/importData.go (hunk 5 above)
Comment (ShivanshGahlot) [reply yb-voyager/cmd/importData.go:1038]: Oh yeah my bad. Realised it later that `getTargetSchema` returns public in case of no schema name.

Comment (ShivanshGahlot) [reply yb-voyager/cmd/importData.go:1038]: Oh yeah my bad. I was applying a condition inside the for loop before to handle the case when schema name is not attached but then realised that getTargetSchema returns public in that case as well. Forgot to change this line.

File: yb-voyager/cmd/importData.go (hunk 1 above)
Comment (ShivanshGahlot) [reply yb-voyager/cmd/importData.go:1070]: Done

File: yb-voyager/cmd/importData.go (hunk 4 above)
Comment (ShivanshGahlot) [reply yb-voyager/cmd/importData.go:1035]: Right

File: yb-voyager/cmd/importData.go (hunk 2 above)
Comment (ShivanshGahlot) [reply yb-voyager/cmd/importData.go:1026]: Yep

**Review by amit-yb — APPROVED** [review]
LGTM.

**Pull request merged**